assets/js/client.js

```diff
@@ -125,6 +125,132 @@ document.addEventListener('DOMContentLoaded', function () {
   debugButton.addEventListener('mousedown', onMouseDown);
   debugButton.addEventListener('click', onClick);
 
+  // Highlighting feature
+  const isElementVisible = (element) => {
+    if (!element) return false;
+
+    const style = window.getComputedStyle(element);
+    return (
+      style.display !== 'none' &&
+      style.visibility !== 'hidden' &&
+      style.opacity !== '0'
+    );
+  };
+
+  const createHighlightElement = (activeElement, detail, id) => {
+    const rect = activeElement.getBoundingClientRect();
+    const highlight = document.createElement('div');
+    highlight.id = id;
+    highlight.dataset.attr = detail.attr;
+    highlight.dataset.val = detail.val;
+
+    highlight.style.position = 'absolute';
+    highlight.style.top = `${rect.top + window.scrollY}px`;
+    highlight.style.left = `${rect.left + window.scrollX}px`;
+    highlight.style.width = `${activeElement.offsetWidth}px`;
+    highlight.style.height = `${activeElement.offsetHeight}px`;
+    highlight.style.backgroundColor = '#87CCE880';
+    highlight.style.zIndex = '10000';
+    highlight.style.pointerEvents = 'none';
+
+    return highlight;
+  };
+
+  const highlightElementID = 'live-debugger-highlight-element';
+  const highlightPulseElementID = 'live-debugger-highlight-pulse-element';
+  let activeElement;
+
+  window.addEventListener('phx:highlight', (msg) => {
+    let highlightElement = document.getElementById(highlightElementID);
+
+    if (highlightElement) {
+      highlightElement.remove();
+      if (
+        msg.detail.val === undefined ||
+        highlightElement.dataset.val === msg.detail.val
+      ) {
+        return;
+      }
```

Comment:
Maybe like this? If you don't like it, feel free to ignore
```suggestion
      const cond = msg.detail.val === undefined || highlightElement.dataset.val === msg.detail.val
      if (cond) return;
```